regparser/tree/xml_parser/reg_text.py

```diff
@@ -107,14 +125,23 @@ def build_subpart(reg_part, subpart_xml):
     for ch in subpart_xml.getchildren():
         if ch.tag == 'SECTION':
             sections.extend(build_from_section(reg_part, ch))
-        elif ch.tag == 'SUBJGRP':
-            for group_child in ch.getchildren():
-                if group_child.tag == 'SECTION':
-                    sections.extend(build_from_section(reg_part, group_child))
 
     subpart.children = sections
     return subpart
 
+def build_subjgrp(reg_part, subjgrp_xml):
+    # This handles subjgrps that have been pulled out and injected into the same
+    # level as subparts.
+    subjgrp_title = get_subjgrp_title(subjgrp_xml)
+    subjgrp = reg_text.build_subjgrp(subjgrp_title, reg_part)
+
+    sections = []
+    for ch in subjgrp_xml.getchildren():
+        if ch.tag == 'SECTION':
+            sections.extend(build_from_section(reg_part, ch))
+
+    subjgrp.children = sections
+    return subjgrp
```

Comment:
Similarly, this mirrors the `build_subpart` fn quite a bit. Could they be merged or share a common kernel?